activation/activation.go

```diff
@@ -261,19 +261,28 @@ func (b *Builder) run(ctx context.Context) {
 	b.loop(ctx)
 }
 
-func (b *Builder) generateProof(ctx context.Context) error {
-	// don't generate the commitment every time smeshing is starting, but once only.
-	if _, err := b.cdb.GetLastAtx(b.nodeID); err != nil {
-		// Once initialized, run the execution phase with zero-challenge,
-		// to create the initial proof (the commitment).
-		startTime := time.Now()
-		b.initialPost, b.initialPostMeta, err = b.postSetupProvider.GenerateProof(ctx, shared.ZeroChallenge)
-		if err != nil {
-			return fmt.Errorf("post execution: %w", err)
-		}
-		metrics.PostDuration.Set(float64(time.Since(startTime).Nanoseconds()))
+func (b *Builder) generateInitialPost(ctx context.Context) error {
+	// Generate the initial POST if we don't have an ATX...
+	if _, err := b.cdb.GetLastAtx(b.nodeID); err == nil {
+		return nil
+	}
+	// ...and if we don't have an initial POST persisted already.
+	if _, err := loadPost(b.nipostBuilder.DataDir()); err == nil {
+		return nil
 	}
 
+	// Create the initial post and save it.
+	startTime := time.Now()
+	var err error
+	b.initialPost, b.initialPostMeta, err = b.postSetupProvider.GenerateProof(ctx, shared.ZeroChallenge)
+	if err != nil {
+		return fmt.Errorf("post execution: %w", err)
+	}
+	metrics.PostDuration.Set(float64(time.Since(startTime).Nanoseconds()))
+
+	if err := savePost(b.nipostBuilder.DataDir(), b.initialPost); err != nil {
+		b.log.With().Warning("failed to save initial post: %w", log.Err(err))
```

Comment:
I was thinking about whether I should return an error, panic, or log and continue. I picked the latter because it's not really a fatal error and the application _can_ continue. Terminating would mean that the post needs to be regenerated and there's really no guarantee that `savePost()` would succeed on the second try.

Reply:
but that failure means something is not ok with os or disk, e.g can't fit the data for example